translator/src/translator.rs

```diff
@@ -0,0 +1,450 @@
+                if symbol_table.file_path.as_path().ends_with("builtins.pyi") {
+                    details.name.to_string()
+                } else {
+                    format!("{}::{}", get_module_name(symbol_table.file_path.as_path()), &details.name)
+                }
+            }
+        }
+    }
+}
+
+impl<'a> TraversalVisitor for CppTranslator<'a> {
+    fn visit_stmt(&mut self, s: &ast::Statement) {
+        self.write_indent();
+        match s {
+            Statement::ExpressionStatement(e) => self.visit_expr(e),
+            Statement::Import(i) => self.visit_import(i),
+            Statement::ImportFrom(i) => self.visit_import_from(i),
+            Statement::AssignStatement(a) => {
+                self.visit_assign(a);
+                writeln!(self.output, ";");
+            },
+            Statement::AnnAssignStatement(a) => self.visit_ann_assign(a),
+            Statement::AugAssignStatement(a) => self.visit_aug_assign(a),
+            Statement::Assert(a) => self.visit_assert(a),
+            Statement::Pass(p) => self.visit_pass(p),
+            Statement::Delete(d) => self.visit_delete(d),
+            Statement::ReturnStmt(r) => {
+                self.visit_return(r);
+                writeln!(self.output, ";");
+            },
+            Statement::Raise(r) => self.visit_raise(r),
+            Statement::BreakStmt(b) => self.visit_break(b),
+            Statement::ContinueStmt(c) => self.visit_continue(c),
+            Statement::Global(g) => self.visit_global(g),
+            Statement::Nonlocal(n) => self.visit_nonlocal(n),
+            Statement::IfStatement(i) => self.visit_if(i),
+            Statement::WhileStatement(w) => self.visit_while(w),
+            Statement::ForStatement(f) => self.visit_for(f),
+            Statement::WithStatement(w) => self.visit_with(w),
+            Statement::TryStatement(t) => self.visit_try(t),
+            Statement::TryStarStatement(t) => self.visit_try_star(t),
+            Statement::FunctionDef(f) => self.visit_function_def(f),
+            Statement::ClassDef(c) => self.visit_class_def(c),
+            Statement::MatchStmt(m) => self.visit_match(m),
+            Statement::AsyncForStatement(f) => self.visit_async_for(f),
+            Statement::AsyncWithStatement(w) => self.visit_async_with(w),
+            Statement::AsyncFunctionDef(f) => self.visit_async_function_def(f),
+            Statement::TypeAlias(a) => self.visit_type_alias(a),
+        }
+    }
+
+    fn visit_expr(&mut self, e: &Expression) {
+        match e {
+            Expression::Constant(c) => self.visit_constant(c),
+            Expression::List(l) => self.visit_list(l),
+            Expression::Tuple(t) => self.visit_tuple(t),
+            Expression::Dict(d) => self.visit_dict(d),
+            Expression::Set(s) => self.visit_set(s),
+            Expression::Name(n) => self.visit_name(n),
```

Comment:
what is the name used for, could it be for both classes and functions?

Reply:
Yeah, it comes up in a variety of contexts -- module names, class names, function names, maybe more that I can't think of right now :)

Reply:
could it potentially overlap with the Call if it can be a function name? maybe the expression checks () and can distinguish?